### DISCUSSION TITLE
How to repeat noise in given time?

### DISCUSSION THREAD

**Discussion by tgrajewski** [stegu/psrdnoise#7] [Q&A] [answered]
It seems that all examples and tutorials use incrementing `time` uniform, which grows to infinity slowly.

Is there a way to wrap the `time` value, so that it the noise repeats itself seamlessly after N seconds?

**Comment by redblobgames** [comment]
I think many of the demos use `time` as the _rotation_ parameter, so it's repeating automatically. But if you're not using time for rotation, I think one easy way would be to use `sin(time)` instead of `time`.

**Comment by stegu** [comment, marked as answer]
How to make the rotations repeat depends on how you use the 'time'
parameter. You must find a common period for all rotations and wrap the
uniform input value to that.
If one component of noise uses 2*pi*time for the 'alpha' rotation, another
2*pi*time*0.1 and a third uses 2*pi*time/15.0, all expressions will attain
an even multiple of 2*pi at time=30, thus you can send a uniform 'time'
that wraps at 30 seconds to avoid the bad precision with large values for
'time'.
If you use factors that have common primes, like time*2*pi/2^k, 0<=k<=N,
all rotations will wrap at time=2^N. Noise is often used as a sum of
components that scale by some power series, 2^k or sqrt(2)^k, both in terms
of the spatial and the temporal scaling (size and speed).
